pytorch_lightning/utilities/cli.py

```diff
@@ -332,6 +347,15 @@ def add_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
 
     def link_optimizers_and_lr_schedulers(self) -> None:
         """Creates argument links for optimizers and learning rate schedulers that specified a ``link_to``."""
+        if any(
+            True for v in sys.argv for optim_name in OPTIMIZER_REGISTRIES.keys() if f"--optimizer={optim_name}" in v
+        ):
+            self.parser.add_optimizer_args(self.optimizer_registered)
```

Comment:
The users opt-into adding optimizers/schedulers by overriding `add_arguments_to_parser`. Do you plan to change that entirely?

Reply:
Added a check for lr_scheduler, optimizer group existence.